server.js

```diff
@@ -53,51 +53,31 @@ app.post('/api/user', (req, res) => {
     // It is good practice to specifically pick the fields we want to insert here *in the backend*,
     // even if we have already done so on the front end. This is to prevent malicious users
     // from adding unexpected fields by modifying the front end JS in the browser.
-
     var newUser =  _.pick(req.body, [
         'name', 'email', 'phone', 'passphrase'])
-    newUser.isAdmin = false
-    newUser.approvedBy = ''
-    bcrypt.hash(newUser.passphrase, 10, (err, hash) => {
-        // Store hash in database
-        newUser.passphrase = hash
-        db.insertOne('user', newUser).then(result => {
-            return res.json(result)
-        }).catch(error => {
-            console.log(error)
-            return res.status(422).json(error)
-        })
-    })
-})
-
-app.post('/api/posts', function(req, res) {
-    delete req.body['_id']
-    db.insertOne(posts_db_name, req.body).then(response => {
-        res.status(201).json(response);
-    }).catch(error => {
-        res.status(406).json({'error': error})
-    })
-})
-
-
-app.put('/api/posts', function(req, res) {
-    db.updateOneById(posts_db_name, req.body).then(response => {
-        res.status(201).json({msg: 'successfully edited report'});
+    createNewUser(newUser).then(result => {
+        return res.json(result)
```

Comment:
If an error happens in `createNewUser`, you'll still return a successful JSON result.

We might want to move the `.catch` function in `createNewUser` to here instead, and then return a JSON result with an HTTP error code.

Reply:
+1